pkg/resolver/search.go

```diff
@@ -495,13 +526,29 @@ func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
+				// If property is of array type like label, remove the equal sign in it and use colon
+				// - to be similar to how it is stored in the database
+				if _, ok := arrayProperties[filter.Property]; ok {
+					cleanedVal := make([]string, len(values))
+					for i, val := range values {
+						labels := strings.Split(val, "=")
+						if len(labels) > 1 {
+							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
+						} else {
+							//// If property is of array type, format it as an array for easy searching
+							cleanedVal[i] = labels[0]
+						}
```

Comment:
A string like `label=a=b` will cause problems here. Let's update this to at least log this problem.
```suggestion
						if len(labels) == 2 {
							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
						} else if len(labels) == 1{
							//// If property is of array type, format it as an array for easy searching
							cleanedVal[i] = labels[0]
						} else {
							klog.error("Error while decoding label string")
							cleanedVal[i] = val // <<<REVIEW NOTE: is this the best we can do in this situation?
						}
```

Reply:
Done.